modules/apps/27-interchain-accounts/controller/keeper/handshake.go

```diff
@@ -162,30 +187,49 @@ func (k Keeper) OnChanUpgradeInit(ctx sdk.Context, portID, channelID string, ord
 		return "", err
 	}
 
-	if err := icatypes.ValidateControllerMetadata(ctx, k.channelKeeper, connectionHops, metadata); err != nil {
-		return "", errorsmod.Wrap(err, "invalid metadata")
+	// ValidateControllerMetadata will ensure the interchain account version has not changed and that the
```

Comment:
we could pull here down into a helper func `ValidateUpgradeMetadata`, pulling out the relevant checks from `ValidateControllerMetadata`, but left as duplication between upgrade init and upgrade ack for now